pkg/operator/cassandra/nodetool/client/client.go

```diff
@@ -0,0 +1,92 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+
+	"github.com/rook/rook/pkg/operator/cassandra/nodetool/version"
+)
+
+const (
+	storageServicePath = "read/org.apache.cassandra.db:type=StorageService"
+)
+
+type StorageService struct {
+	HostIdMap        map[string]uuid.UUID `json:""`
+	LiveNodes        []string             `json:""`
+	UnreachableNodes []string             `json:""`
+	LeavingNodes     []string             `json:""`
+	JoiningNodes     []string             `json:""`
+	MovingNodes      []string             `json:""`
+	LocalHostId      uuid.UUID            `json:""`
+	ReleaseVersion   *version.Version     `json:""`
+}
+
+type Interface interface {
+	StorageService() (*StorageService, error)
+}
+
+type client struct {
+	storageServiceURL *url.URL
+	client            *http.Client
+}
+
+var _ Interface = &client{}
+
+func New(baseURL *url.URL, c *http.Client) Interface {
+	storageServiceURL, err := url.Parse(storageServicePath)
+	if err != nil {
+		panic(err)
```

Comment:
is panic the right thing to do here? for a function that is just supposed to return an interface, I would expect it to be up to the caller on if they want to panic or handle it some other way